### PR TITLE
feature/js_respect-order

### PR DESCRIPTION
**Pull request by jsanderson44**
Changed hints dictionary to be an array of tuples that are sorted alphabetically by key so that the order is respected when generating file

### CODE REVIEW

File: configen/OptionsParser.swift [hunk 1]
```diff
@@ -71,7 +69,7 @@ final class OptionsParser {
       let (variableName, type) = (hints[0], hints[1])
       hintsDictionary[variableName] = type
     }
-
-    return hintsDictionary
+    
+    return hintsDictionary.sorted(by: { $0.0 < $1.0 })
```
Comment (theblixguy) [inline configen/OptionsParser.swift]: You can write this in point-free style: `hintsDictionary.sorted(by: <)`

File: configen/OptionsParser.swift [hunk 2]
```diff
@@ -54,14 +53,13 @@ final class OptionsParser {
 
     return plistDictionary
     }()
-
-  lazy var hintsDictionary: [String: String] = { [unowned self] in
+  
+  lazy var sortedHintsTuple: [(key: String, value: String)] = { [unowned self] in
```
Comment (theblixguy) [inline configen/OptionsParser.swift]: We aren't using the explicit labels anywhere so perhaps we can drop this (or change existing code to use it)

File: configen/OptionsParser.swift (hunk 1 above)
Comment (jsanderson44) [reply configen/OptionsParser.swift]: Nice. Updated

File: configen/OptionsParser.swift (hunk 2 above)
Comment (jsanderson44) [reply configen/OptionsParser.swift]: Changed to use a typealias. Let me know what you think 👍

File: configen/OptionsParser.swift [hunk 3]
```diff
@@ -69,7 +71,6 @@ final class OptionsParser {
       let (variableName, type) = (hints[0], hints[1])
       hintsDictionary[variableName] = type
     }
-    
-    return hintsDictionary.sorted(by: <)
+    return hintsDictionary.sorted(by: <).map { Hint(variableName: $0.key, type: $0.value) }
```
Comment (theblixguy) [inline configen/OptionsParser.swift]: You can drop the `map` if you change the `hintsDictionary` to `[Hint]`. Up to you!

**Review by theblixguy — APPROVED** [review]

File: configen/OptionsParser.swift (hunk 2 above)
Comment (theblixguy) [reply configen/OptionsParser.swift]: Nice 👌

File: configen/OptionsParser.swift (hunk 3 above)
Comment (jsanderson44) [reply configen/OptionsParser.swift]: Nice. Have updated

**Review by theblixguy — APPROVED** [review]
LGTM!

File: configen/OptionsParser.swift [hunk 4]
```diff
@@ -54,24 +55,21 @@ final class OptionsParser {
 
     return plistDictionary
     }()
-
-  lazy var hintsDictionary: [String: String] = { [unowned self] in
+  
+  lazy var sortedHintsTuple: [Hint] = { [unowned self] in
```
Comment (bengilroy) [inline configen/OptionsParser.swift]: Thoughts on renaming this to `sortedHints`? Just because this is an array of tuples, not a tuple

**Review by bengilroy — APPROVED** [review]
Looks good just one suggestion

File: configen/OptionsParser.swift (hunk 4 above)
Comment (jsanderson44) [reply configen/OptionsParser.swift]: Ah yeah. Have updated. Thanks

**Pull request merged**